packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -114,6 +147,23 @@ abstract class ApplicationBase extends cdk.Resource implements IApplication {
     }
   }
 
+  /**
+   * Create an attribute group and associate this application with the created attribute group.
+   */
+  public addAttributeGroup(props: AttributeGroupAssociationProps): IAttributeGroup {
+    const attributeGroup = new AttributeGroup(this, props.attributeGroupLogicalId, {
```

Comment:
This is not a logical ID, it's a construct ID. Please at least make it optional?

```suggestion
    const attributeGroup = new AttributeGroup(this, props.constructId ?? 'Attributes', {
```

Reply:
Removed it and calculate it from the attribute group name.